src/cache/cacher.rs

```diff
@@ -41,9 +64,42 @@ impl RedisCache {
     /// # Arguments
     ///
     /// * `url` - It takes an url as a string.
-    pub fn cached_json(&mut self, url: &str) -> Result<String, Box<dyn std::error::Error>> {
-        let hashed_url_string = Self::hash_url(url);
-        Ok(self.connection.get(hashed_url_string)?)
+    pub async fn cached_json(&mut self, url: &str) -> Result<String, Report<PoolError>> {
+        self.current_connection = Default::default();
```

Comment:
this will always reset `current_connection` to 0, this means the same connection will always be used, and others will be ignored

Reply:
I think to understand why I did this, I think I will first explain in brief what the code does. So the code actually what it does is that when a connection from the pool let's call this A fails due to `connection drop error` then this is replaced with another connection B from the pool. Now, Since I am using the connection manager feature, connection A is automatically reconnected in the background, which can take sometime. Which means that if that connection A were to be used, then we would have to wait for it to be re-connected. Which means it will add a time delay (or latency) in some cases it would be a significant delay which is bad you know. Also, there is another scenario where you might even have a situation where the connection drops repeatedly because of some unknown reason. Maybe the cloud service provider who provided the Redis service has some issue or maybe due internet connection lose. That's why what the code does is it replaces A with B. Now what I have done is that when you search the next time then connection A would have been re-connected which means we could re-use it. So that's why I have added the line which you have pointed above. :slightly_smiling_face: